### DISCUSSION TITLE
New Setup Problems

### DISCUSSION THREAD

**Discussion by Nathan-CPSD** [AdhocAdam/smletsexchangeconnector#259] [Q&A] [answered]
Good Day,

Hope that I am not placing this in the wrong location, but did not want to make an issue unless it was needed.  I noted that Discussion could be made into Issues, so figured I would start here. 

We have been using he MS Exchange Connector for some time, but recently it has caused us issues and I come upon the smletsexchangeconnector and moving over to it was a no brainer. 

Current configuration:
SMLETS is installed.
AD-PS installed
Microsoft Exchange Web Services DLL in SCSM Directory. v. 15.00.0913.015
smletsexchangeconnector DLL in SCSM Directory
Management Pack Installed and configured using wiki. 
 *as a note the old exchange connector is still installed and running until I have confirm the new one. 

I have not been able to get the new connector to ingest any messages, I have set up a new inbox to not disturb the current connector and have a smooth cut over. 

The only logs that have been created are in the Operations Manager Log with the following error:

A Windows Workflow Foundation workflow failed during execution.
 
Workflow Type: SMLets.Exchange.Connector.Resources.RunScript 

Workflow Identifier: 7bdb4c11-31b8-bf66-e2ef-54de5d109462 

Exception Type: System.Management.Automation.RuntimeException 

Exception Message: Method invocation failed because [Microsoft.Exchange.WebServices.Data.ImpersonatedUserId] does not contain a method named 'new'. 

Exception Stack:    at CallSite.Target(Closure , CallSite , RuntimeType , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at System.Management.Automation.Interpreter.DynamicInstruction`4.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)

My first though was to remove and re-add the MP and restart the configuration. 

I have looked though the open and solved issue and did not see this issue come up in the past. 

Thank you for taking the time to look at this and for developing this much needed tool for SCSM. 

**Update** I have continued working though out the day and I was able to narrow down the first issue to either a bad password on a run-as account or the use of impersonation. I created a new run-as account for the new mailbox and have been able to successful get new connector to work.

I have come across two new errors, the first being when I have the multi-mailbox settings setup (Currently testing with two) I receive the following error: 
A Windows Workflow Foundation workflow failed during execution.
 
Workflow Type: SMLets.Exchange.Connector.Resources.RunScript 

Workflow Identifier: 7bdb4c11-31b8-bf66-e2ef-54de5d109462 

Exception Type: System.Management.Automation.MethodException 

Exception Message: Cannot find an overload for "add" and the argument count: "2". 

Exception Stack:    at CallSite.Target(Closure , CallSite , Object , String , Hashtable )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at System.Management.Automation.Interpreter.DynamicInstruction`4.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)

When I remove one of the mailboxes and only keep one in the multi configuration it will pick up the mail but the SR's that are created do not have the templates set that are designated, as well there is no affected or assigned user.

**Comment by AdhocAdam** [comment, marked as answer]
Greetings @Nathan-CPSD!

From the looks of things:
- You're trying to use your chosen Run As Account to Impersonate the Email Address you typed in
- The workflow is successfully kicking off the smletsexchangeconnector.ps1 file, because the Exception message sounds a whole lot like line 4064 in the PowerShell: https://github.com/AdhocAdam/smletsexchangeconnector/blob/0f3edf7aae77b9c7e29830d473123e5df39aefc6/smletsExchangeConnector.ps1#L4061-L4065
- I don't have a logging event for when line 4064 runs, so the event ends up in Operations Manager as opposed to the SMLets Exchange Connector log

What's strange here is that the error says "new" doesn't exist on the Exchange Impersonation Type...which I don't know how that's possible. Take the following 3 line PowerShell example. EWS Path is defined, loaded, and then begin typing out one of the Exchange types in VSCode (or ISE if you prefer):

![impersonateExample](https://user-images.githubusercontent.com/6636040/105247730-db469980-5b3a-11eb-9391-da22402853fb.PNG)


Never the less, I'll see if there is someway I can try and reproduce this. Until then, would it be possible for you to:
- Disable Impersonation
- Setup the account/inbox you created as an SCSM Run As Account so that the Run As Account/Email Address you typed in matches?

**Reply by AdhocAdam** [reply]
Oh! Just saw your update as I posted. Lemme read that over.

**Reply by AdhocAdam** [reply]
There is actually an open Issue on this that I've been trying to get back to/figure out how to move forward on. Check out [212](https://github.com/AdhocAdam/smletsexchangeconnector/issues/212).

If you can shed some light on this given your current setup it would absolutely go towards addressing the discrepancy on the issue!

**Reply by AdhocAdam** [reply]
This ultimately appears to be tied to #212. It's fix will be included via #261 with v3.2 of the connector on #257
